self-core-impl/src/test/java/com/selfxdsd/core/SelfCoreTestCase.java

```diff
@@ -107,5 +111,33 @@ public void closesStorageOnClose() throws Exception {
         Mockito.verify(storage, Mockito.times(1)).close();
     }
 
+    /**
+     * Should verify an authenticated user.
+     */
+    @Test
+    public void authenticatedUser() {
+        final User user = Mockito.mock(User.class);
+        Mockito.when(user.username()).thenReturn("john");
+        Mockito.when(user.email()).thenReturn("john@john.com");
+        final Provider provider = Mockito.mock(Provider.class);
+        Mockito.when(user.provider()).thenReturn(provider);
+        final Projects all = Mockito.mock(Projects.class);
+        Mockito.when(user.projects()).thenReturn(all);
+        final Authenticated authenticated = new Authenticated(user, "tok3n");
+        final Provider prov = new Authenticated(user, "tok3n").provider();
 
+        MatcherAssert.assertThat(
+            authenticated.username(),
+            Matchers.equalTo("john")
+        );
+        MatcherAssert.assertThat(
+            authenticated.email(),
+            Matchers.equalTo("john@john.com")
+        );
+        Mockito.verify(provider).withToken(anyString());
+        MatcherAssert.assertThat(
+            authenticated.projects(),
+            Matchers.equalTo(all)
+        );
+    }
```

Comment:
@ANadrowski This test is overall OK, but it tests the whole ``Authenticated`` class (all of its methods). A unit test should cover only one method. 

Can you please make a new test class, called ``AuthenticatedTestCase`` and add 4 tests? One for username, one for email, one for provider/withTocken and one for projects? 

Just beak down this big test into 4 smaller ones. For instance, in the case of ``Authenticated.username()`` you do not have to mock any ``Projects``.

Reply:
@amihaiemil yeah, it makes sense. I'll implement it :)